crates/ark/src/interface.rs

```diff
@@ -423,8 +428,13 @@ impl RMain {
         if !ignore_user_r_profile {
             startup::source_user_r_profile();
         }
+    }
 
-        // Does not return!
+    /// Start the REPL. Does not return!
+    pub fn start() {
+        // Update the main thread ID in case the REPL is started in a background
+        // thread (e.g. in integration tests)
+        unsafe { R_MAIN_THREAD_ID = Some(std::thread::current().id()) };
```

Comment:
Should we even set `R_MAIN_THREAD_ID` from `setup()` then? It feels weird to do so twice to me.

Reply:
I feel the same but we have to because `write_console()`, called during startup when the banner is emitted, calls `RMain::get_mut()`.

Reply:
I've now:

- Split `write_console()` into pre-init and post-init handlers, as you suggested yesterday. The pre-init handler pushes output to a new global variable `R_BANNER` so it doesn't need to call `RMain::get_mut()`.

- Added the `RMain::positron_ns` field. The modules watchers now call `RMain::get_mut()` to access the namespace. This solves issues having to do with the creation of an `RThreadSafe` object which requires `RMain::get()` which would fail if the R thread ID is not initialized.

This makes it possible to set up the thread ID only once, in `RMain::start()`.